{{ cookiecutter.__project_name_kebab }}/.github/workflows/nightly.yml

```diff
@@ -0,0 +1,30 @@
+name: Nightly Wagtail Test
+
+on:
+  schedule:
+    - cron: '0 1 * * *'
+    # At 01:00, daily
+  workflow_dispatch:
+
+jobs:
+  nightly-wagtail-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - run: git clone https://github.com/wagtail/wagtail.git
+
+      - run: python -m pip install flit
+      - run: flit install --deps production --extras testing
+      - run: python -m pip install ./wagtail
+
+      - run: python testmanage.py test
+
+      - name: Report failure
+        run: |
+          python -m pip install requests
+          python ./.github/scripts/report_nightly_build_failure.py
+        if: failure()
```

Comment:
report_nightly_build_failure requres SLACK_WEBHOOK_URL set in env, so this will fail, and I would rather we don't try to run something that just won't work..

Perhpas add a note about https://github.com/wagtail/wagtail/wiki/Nightly-testing-of-official-plugins and keep the whole block commented?

Reply:
If the webhook URL is not set, it does have a fallback to just print an error to the console:

https://github.com/wagtail/cookiecutter-wagtail-package/blob/781a73739616ecca7b8a1fd9666822dc50e244b1/%7B%7B%20cookiecutter.__project_name_kebab%20%7D%7D/.github/scripts/report_nightly_build_failure.py#L23-L26

We could run this step conditionally on existence of the secret if you want. We definitely should document `SLACK_WEBHOOK_URL` more clearly.

Reply:
Here's a quick example of how to run a workflow step conditional on existance of a secret:
- https://github.com/badges/shields/blob/0c73b3591562d8478966299c035dfaba2059504c/.github/workflows/test-integration-17.yml#L13-L14
- https://github.com/badges/shields/blob/0c73b3591562d8478966299c035dfaba2059504c/.github/workflows/test-integration-17.yml#L38-L42

Reply:
ah I missed https://github.com/wagtail/cookiecutter-wagtail-package/blob/781a73739616ecca7b8a1fd9666822dc50e244b1/%7B%7B%20cookiecutter.__project_name_kebab%20%7D%7D/.github/scripts/report_nightly_build_failure.py#L23-L26

I guess chaging `if: failure()` to something `if: ${{ failure() && secrets.SLACK_WEBHOOK_URL != '' }}` should take care of: not running the script unnecessarily, serve as an indication that it requires a `SLACK_WEBHOOK_URL` env var, be it the official one, or an internal one

Reply:
You can't use secrets directly in a conditional. See
https://github.com/orgs/community/discussions/26726
https://github.com/actions/runner/issues/520
that's why I assign it to an intermediate env var in the example I showed, but.. yeah, basically.

I have also just realised having given this a second look that I might need to explicitly pass `secrets.SLACK_WEBHOOK_URL` in the env when I invoke `report_nightly_build_failure.py`. Need to give this another test in my other repo.